composeApp/src/commonMain/kotlin/org/ooni/engine/Engine.kt

```diff
@@ -40,6 +64,73 @@ class Engine(
             }
         }
 
+    fun session(finalSettings: TaskSettings): OonimkallBridge.Session {
+        return bridge.newSession(
+            object : OonimkallBridge.SessionConfig {
```

Comment:
You can use the data class constructor now.

Reply:
Session config is still an interface.

Reply:
Ups, then I missed that one. Can you turn it into a `data class` as well?

Reply:
Done that already. Testing the code now.